### ISSUE TITLE
[Package Issue]: Subtitle Edit

### ISSUE THREAD

**Issue by darkred** [microsoft/winget-pkgs#95792]
### Please confirm these before moving forward

- [X] I have searched for my issue and not found a work-in-progress/duplicate/resolved issue.
- [X] I have not been informed if the issue is resolved in a preview version of the winget client.

### Category of the issue

Installer hash mismatch.

### Brief description of your issue

Winget is showing version 3.6.10.0  of installed Subtitle Edit but installed version is 3.6.11.0 . Therefore it's showing incorrect version.
Also, cc @niksedk , Subtitle Edit's developer.

### Steps to reproduce

Install Subtitle Edit (latest version [SubtitleEdit-3.6.11](https://github.com/SubtitleEdit/subtitleedit/releases/download/3.6.11/SubtitleEdit-3.6.11-Setup.zip)) on win 11 x64 22H2 *(you may also uninstall and reinstall SE again)*.

### Actual behavior

Winget shows version 3.6.10.0 as installed version and that there's 3.6.11.0 available.

### Expected behavior

It should recognize the installed version 3.6.11.0 instead, therefore not list Subtitle Text at all.

### Environment

```shell
Windows: Windows.Desktop v10.0.22621.1194
System Architecture: X64
Package: Microsoft.DesktopAppInstaller v1.19.10173.0

Logs: %LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\DiagOutputDir
```

### Screenshots and Logs

![2023-02-07_121347](https://user-images.githubusercontent.com/723651/217216971-ed24126e-d48a-463c-a9f0-1361c30946f0.jpg)
[WinGet-2023-02-07-12-11-39.389.zip](https://github.com/microsoft/winget-pkgs/files/10673914/WinGet-2023-02-07-12-11-39.389.zip)

**Comment by mdanish-kh** [comment]
It's because the version `3.6.11.0` installs side-by-side with `3.6.10.0` or previous versions. I've checked the previous versions and they all upgrade 'in-place', so this is a recent issue.
```pwsh
~ winget list subtitle
Name                 Id                 Version  Available Source
-----------------------------------------------------------------
Subtitle Edit        Nikse.SubtitleEdit 3.6.11.0           winget
Subtitle Edit 3.6.10 Nikse.SubtitleEdit 3.6.10.0 3.6.11.0  winget
```
This should be fixed by adding `UpgradeBehavior: uninstallPrevious` in 3.6.11.0's manifest to tell WinGet to uninstall the previous version first but it doesn't seem to call the uninstaller when I locally test the manifest on my machine.

If you want a quicker fix on your end, you can manually uninstall the previous version from `appwiz.cpl`

**Comment by mdanish-kh** [comment]
So the reason why WinGet is not able to call the uninstaller before installing the newer version is that WinGet is trying to match the new installer with the previous installation to get a match and call the uninstaller. The issue is that `3.6.10.0` and previous versions write their uninstall entries to `Computer\HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\SubtitleEdit_is1` while the newer version writes to `Subtitle Edit_is1` (notice the whitespace), so WinGet is not able to get a match and call the uninstaller.

This should be brought up to the publisher to fix.

**Comment by mdanish-kh** [comment]
[Policy] Installer-Issue

**Comment by mdanish-kh** [comment]
Close with reason: Package writes uninstall entries to correct location now;

The ProductCode got fixed in the latest `3.6.12.0` version  (https://github.com/microsoft/winget-pkgs/pull/100740) by the publisher (Thanks @niksedk), so hopefully you won't see the problem in future upgrades (if ProductCode remains the same). You still might have to manually uninstall `3.6.11.0`.
